auto_gptq/nn_modules/qlinear/qlinear_cuda.py

```diff
@@ -268,8 +268,8 @@ def forward(self, x: torch.Tensor):
                     g_idx_i = self.g_idx[i*num_dim:(i+1)*num_dim]
                     weights.append(scale_i[g_idx_i.long()] * (weight_i - zeros_i[g_idx_i.long()]))
                 weights = torch.cat(weights,dim=1)
-            out = torch.matmul(x, weights)
-        out = out.to(dtype=weights.dtype).reshape(out_shape)
+            out = torch.matmul(x, weights).to(dtype=weights.dtype)
```

Comment:
I'm not sure to remember why these casts are needed in the first place. Shouldn't the activation & weight be of same dtype (either both fp32, either both fp16)?